src/app.js

```diff
@@ -21,23 +21,25 @@ export default function (options) {
 		}
 	}, options.hooks)
 
-	for (var name in merge(reducers, effects)) {
-		(function (name) {
-			actions[name] = function (data) {
-				hooks.onAction(name, data)
-
-				var effect = effects[name]
-				if (effect) {
-					return effect(model, actions, data, hooks.onError)
-				}
-
-				var update = reducers[name], _model = model
-				render(model = merge(model, update(model, data)), view, node)
-
-				hooks.onUpdate(_model, model, data)
-			}
-		}(name))
+	function recurseActions (isReducer, pointer, actionPointer, _name) {
```

Comment:
I'm really sure that we can't avoid that. The signature is also as good as possible AFAIK.

I agree that the signature isn't the most readable one, but try removing anything. There is nothing that isn't necessary.